### PR TITLE
Improvements to wsdl2apigee tool

### PR DESCRIPTION
**Pull request by mat-613**
-Added support for oauth policy in passthru proxy based on the parameter already present oauth
-Added support for external xsd file referenced into the wsdl file 
-Added new parameter backendurl in order to specify the backend url of the proxy, when it’s passed to the tool it takes it as the URL of HTTPTargetConnection. 
-Added new parameter backendurlvalidation, a boolean parameter that can disable the url validation for the target, default it’s true.

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -35,13 +35,15 @@ basepath=Uses the basepath from the wsdl
 cors=false                Don't enable CORS; Works only for SOAP to REST
 build=specify build folder   default is temp folder; ensure user has access to read/write to temp folder
 opsmap=specify operations map
-``` 
+backendurl = Specify the target backend url
+backendurlvalidation=false    Specify to disable the backendurl validation
```
Comment (ssvaidyanathan) [inline README.MD:39]: Why is this required?

File: src/main/java/com/apigee/proxywriter/GenerateProxy.java [hunk 2]
```diff
@@ -2407,6 +2457,43 @@ private void getWSDLDetails(String wsdlPath) throws Exception {
 		}.getClass().getEnclosingMethod().getName());
 	}
 
+    private void flatWsdlWithExternalXsdImport(String wsdlPath) {
```
Comment (ssvaidyanathan) [inline src/main/java/com/apigee/proxywriter/GenerateProxy.java:2460]: Can we add some samples to the test folder?

Comment (ssvaidyanathan) [reply src/main/java/com/apigee/proxywriter/GenerateProxy.java:2460]: Also can we format the java file? the Indentation is off

File: README.MD (hunk 1 above)
Comment (mat-613) [reply README.MD:39]: Hi Sai, we need it because for our url that look like this `https://app-name-env.gcp.companyname.companylocation.loc/serviceName` and doesn't pass the url validation

File: src/main/java/com/apigee/proxywriter/GenerateProxy.java (hunk 2 above)
Comment (mat-613) [reply src/main/java/com/apigee/proxywriter/GenerateProxy.java:2460]: Hi @ssvaidyanathan  yes i will format the code and add some samples to test folder!

Comment (mat-613) [reply src/main/java/com/apigee/proxywriter/GenerateProxy.java:2460]: Hi @ssvaidyanathan for the sample what do you mean just the wsdl with an external xsd reference files or also a test with the generation of the bundle?

Comment (ssvaidyanathan) [reply src/main/java/com/apigee/proxywriter/GenerateProxy.java:2460]: just a wsdl with an external xsd should be fine. More of a sample for others to try if needed

Comment (ssvaidyanathan) [reply src/main/java/com/apigee/proxywriter/GenerateProxy.java:2460]: the indentation is still off. Are you planning to fix that too?

**Comment by ssvaidyanathan** [comment]
LGTM
@srinandan - can you pls merge?

**Comment by mat-613** [comment]
any news on it?
